src/foam/mlang/mlang.js

```diff
@@ -2380,7 +2382,9 @@ while ( i.hasNext() ) {
 
   try {
     String s = "";
-    if ( prop instanceof foam.core.AbstractFObjectPropertyInfo ) {
+    if ( ReferenceExpressionHelper.isPropertyAReference((PropertyInfo)prop) ) {
+      if ( checkNestedFObject(REF(prop).f(obj)) ) return true;
```

Comment:
may create infinite loop need to check for references that were checked. or it's overkill and i will close this pr later